keyboards/abko/ak84bt/keymaps/default/keymap.c

```diff
@@ -0,0 +1,180 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "config.h"
+#include <stdarg.h>
+#include <stdio.h>
+//#include "quantum.h"
+#include "gpio.h"
+#include "encoder.h"
+
+#include "spi_master.h"
+
+#include "ak84bt.h"
+
+typedef union {
+  uint32_t raw;
+  struct {
+    bool     rgb_layer_change :1;
+    bool     winlock_state :1;
+  };
+}user_config_t;
+
+user_config_t user_config;
+
+enum layer_names
+{
+    _BASE,
+    _FnLay,
+    _NONE1,
+    _NONE2,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {  
+    [_BASE] = LAYOUT(    
+    KC_ESC  , XXXXXXX  , KC_F1   , KC_F2   , KC_F3   , KC_F4   , KC_F5    , KC_F6   , KC_F7   , KC_F8    , KC_F9   ,   KC_F10  , KC_F11  , KC_F12  , XXXXXXX , KC_END  ,
+    KC_GRV  , KC_1     , KC_2    , KC_3    , KC_4    , KC_5    , KC_6     , KC_7    , KC_8    , KC_9     , KC_0    ,   KC_MINS , KC_EQL  , KC_BSPC , KC_HOME , XXXXXXX ,
+    KC_TAB  , KC_Q     , KC_W    , KC_E    , KC_R    , KC_T    , KC_Y     , KC_U    , KC_I    , KC_O     , KC_P    ,   KC_LBRC , KC_RBRC , KC_BSLS , KC_DEL  , XXXXXXX ,
+    KC_CAPS , KC_A     , KC_S    , KC_D    , KC_F    , KC_G    , KC_H     , KC_J    , KC_K    , KC_L     , KC_SCLN ,   KC_QUOT , XXXXXXX , KC_ENT  , KC_PGUP , XXXXXXX ,
+    KC_LSFT , XXXXXXX  , KC_Z    , KC_X    , KC_C    , KC_V    , KC_B     , KC_N    , KC_M    , KC_COMM  , KC_DOT  ,   KC_SLSH , KC_RSFT , KC_UP   , KC_PGDN , XXXXXXX ,
+    KC_LCTL , KC_LGUI  , KC_LALT , XXXXXXX , XXXXXXX , XXXXXXX , KC_SPC   , XXXXXXX , XXXXXXX , KC_RALT  , MO(_FnLay), KC_RCTL , KC_LEFT , KC_DOWN , KC_RIGHT, XXXXXXX
+    ),
+    [_FnLay] = LAYOUT(
+    KC_RESET_FACTORY , XXXXXXX  , KC_MY_COMPUTER , KC_WWW_HOME, KC_CALCULATOR, KC_MSEL, KC_MPRV, KC_MNXT, KC_MPLY,  KC_MSTP, KC_MUTE, KC_VOLD, KC_VOLU, KC_MAIL, XXXXXXX , XXXXXXX ,
+    XXXXXXX , KC_BT1   , KC_BT2  , KC_BT3  , KC_2D4G , XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX ,   XXXXXXX , XXXXXXX , XXXXXXX , RGB_MOD , XXXXXXX ,
+    XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX , KC_SCRL , KC_INS   , KC_PSCR ,   XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX ,
+    XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX ,   XXXXXXX , KC_BK_LIGHT_HSV_SWITCH, XXXXXXX , XXXXXXX , XXXXXXX ,
+    XXXXXXX , RGB_TOG  , RGB_MOD , XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX , XXXXXXX , XXXXXXX  , XXXXXXX ,   XXXXXXX , XXXXXXX , KC_BK_LIGHTNESS_UP , XXXXXXX , XXXXXXX ,
+    XXXXXXX , KC_WINLOCK_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , KC_BK_LIGHT_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX  , MO(_FnLay), XXXXXXX , KC_BK_LIGHT_SPEED_DOWN , KC_BK_LIGHTNESS_DOWN , KC_BK_LIGHT_SPEED_UP , XXXXXXX
```

Comment:
This will not work.
```suggestion
    XXXXXXX , KC_WINLOCK_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX , XXXXXXX , KC_BK_LIGHT_ONOFF, XXXXXXX , XXXXXXX , XXXXXXX  , _______, XXXXXXX , KC_BK_LIGHT_SPEED_DOWN , KC_BK_LIGHTNESS_DOWN , KC_BK_LIGHT_SPEED_UP , XXXXXXX
```